pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -523,6 +523,15 @@ func (d *Device) Bond() config.Bond {
 	return d.DeviceBond
 }
 
+// Bridge implements the MachineNetwork interface.
+func (d *Device) Bridge() config.Bridge {
```

Comment:
looks like we don't have any validation in `v1alpha1_validation.go` with matching tests for it

we basically we need to do validation as discussed in the original ticket: e.g. same link can't be enslaved twice to a bridge and a bond, etc.

Reply:
or that one shouldn't assign addresses to bridges, etc.

Reply:
I added some validations and tests for them. Feel free to have another look. 

I didn't add

> one shouldn't assign addresses to bridges

because it is a valid use case.